Staging test fails when running yarn hardhat test --network rinkeby

I copied the code from this repo (test/staging/FundMe.staging.test.js) but it doesn't work, this is the error I get:

```shell
AssertionError: expected '60000000000000000' to equal '0'
```

It looks like there is an issue with the endingFundMeBalance variable but I don't know what.
Did someone has the same issue ?

> I copied the code from this repo (test/staging/FundMe.staging.test.js) but it doesn't work, this is the error I get:
> 
> ```shell
> AssertionError: expected '60000000000000000' to equal '0'
> ```
> 
> It looks like there is an issue with the endingFundMeBalance variable but I don't know what. Did someone has the same issue ?

It's likely because your fund transaction gets executed but withdraw transaction failed, so you end up having some ether in your fundMe contract, which is not equal to 0.

I fixed this issue by waiting for one block confirmation after executing the transaction
```shell
const fundTxResponse = await fundMe.fund({ value: sendValue });
await fundTxResponse.wait(1);
const withdrawTxResponse = await fundMe.withdraw();
await withdrawTxResponse.wait(1);
```

--- a/hardhat.config.js
+++ b/hardhat.config.js
@@ -69,4 +69,7 @@ module.exports = {
             1: 0, // similarly on mainnet it will take the first account as deployer. Note though that depending on how hardhat network are configured, the account 0 on one network can be different than on another
         },
     },
+    mocha: {
+        timeout: 500000,
+    },
 }

--- a/test/staging/FundMe.staging.test.js
+++ b/test/staging/FundMe.staging.test.js
@@ -14,8 +14,10 @@ developmentChains.includes(network.name)
           })
 
           it("allows people to fund and withdraw", async function () {
-              await fundMe.fund({ value: sendValue })
-              await fundMe.withdraw();
+              const fundTxResponse = await fundMe.fund({ value: sendValue })
+              await fundTxResponse.wait(1)
+              const withdrawTxResponse = await fundMe.withdraw()
+              await withdrawTxResponse.wait(1)
 
               const endingFundMeBalance = await fundMe.provider.getBalance(
                   fundMe.address

